### PR TITLE
Update the agile development contents in sprint planning and backlog refinement. 

### PR DESCRIPTION
**Pull request by buhongw7583c**
The PR contains updates on two documents under agile development: 
-- Add contents for backlog refinement with goals, impact, measures, guidance and recommend links. 
-- Add contents for sprint planning with recommendation of 'sprint goal' extension on Azure Devops.

### CODE REVIEW

File: agile-development/backlog-management/refinement/readme.md [hunk 1]
```diff
@@ -16,22 +16,41 @@
...
 
 What members of the team should participate?
 
+Ideally all of the team members could participate. This is an ongoing process in which the Product Owner and the Development Team collaborate on the details of Product Backlog items.
+
 ## Impact
 
 What positive impact has been observed as a result of this practice?
 
+The intent of backlog refinement is to ensure that the backlog keeps updating with items that are appropriate to their priority, i.e. higher priority iteams are with more details than lower priority ones . Also its important that at any time there are enough user stories ready for planning in the next or even, where reasonable, the next two or three iterations but not too further away.
```
[the remark this answers is not in the document]
Comment (zoekdestep) [reply agile-development/backlog-management/refinement/readme.md:31]: *nit*: spacing between `ones` and `.`

File: agile-development/backlog-management/refinement/readme.md [hunk 2]
```diff
@@ -16,22 +16,41 @@
...
+
 ## Measures
 
 How might one measure the impact of this ceremony positive or negative?
 
+The expected outcome from the backlog refinement is all team members are on the same page of what they are going to implement in the next or next few iterations with priority. A positive outcome of refinement is one where each backlog item can reasonably be “Done” within the Sprint time - box. Product Backlog items that can be “Done” by the Development Team within
+one Sprint are deemed “Ready” for selection in a Sprint Planning.
```
Comment (zoekdestep) [inline agile-development/backlog-management/refinement/readme.md:38]: *nit*: unnecessary enter

**Review by zoekdestep — CHANGES_REQUESTED** [review]
It's a valuable PR, thanks for contributing! Just a few typo-things that need to be fixed.

**Comment by buhongw7583c** [comment]
> It's a valuable PR, thanks for contributing! Just a few typo-things that need to be fixed.

@zoekdestep Stephanie, thanks so much for the careful review and good catch on my typos and links issue! Fix them and now they should be good. :) Appreciate the time and quick feedback!

**Review by zoekdestep — APPROVED** [review]
Thanks a lot for your contribution Hong! :D

**Comment by buhongw7583c** [comment]
@meulta, can I ask for your kind review for this PR? The PR is to contribute on the agile management's sprint planning and backlog grooming part.  :)

File: agile-development/backlog-management/refinement/readme.md [hunk 3]
```diff
@@ -16,22 +16,40 @@
 
 What are the intended outcomes of the ceremonies?
 
+Backlog refinement or formerly known as [backlog grooming](https://www.agilealliance.org/glossary/backlog-grooming/) is when the [product owner](https://www.agilealliance.org/glossary/product-owner/) and some, or all, of the rest of the team review items on the backlog to ensure the backlog contains the appropriate items, that they are prioritized, and that the items at the top of the backlog are ready for delivery. This activity occurs on a regular basis and may be an officially scheduled meeting or an ongoing activity.
```
Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:19]: what do you think about ready for development over ready for delivery? either way is good, just thinking they have a bit different meanings

File: agile-development/backlog-management/refinement/readme.md [hunk 4]
```diff
@@ -16,22 +16,40 @@
...
 
 What members of the team should participate?
 
+Ideally all of the team members could participate. This is an ongoing process in which the Product Owner and the Development Team collaborate on the details of Product Backlog items.
+
 ## Impact
 
 What positive impact has been observed as a result of this practice?
 
+The intent of backlog refinement is to ensure that the backlog keeps updating with items that are appropriate to their priority, i.e. higher priority items are with more details than lower priority ones. Also it is important that at any time there are enough user stories ready for planning in the next or even, where reasonable, the next two or three iterations but not too further away.
```
Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:31]: nit: not too further away => not too much further

File: agile-development/backlog-management/refinement/readme.md [hunk 5]
```diff
@@ -16,22 +16,40 @@
...
+
 ## Measures
 
 How might one measure the impact of this ceremony positive or negative?
 
+The expected outcome from the backlog refinement is all team members are on the same page of what they are going to implement in the next or next few iterations with priority. A positive outcome of refinement is one where each backlog item can reasonably be “Done” within the Sprint time - box. Product Backlog items that can be “Done” by the Development Team within one Sprint are deemed “Ready” for selection in a Sprint Planning.
```
Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:37]: nit: is all team members => is that all team members

Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:37]: nit: Sprint time - box => sprint time

Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:37]: nit: I think we should go for all lowercase within a sentence - here we have Sprint Planning, Done, Product Backlog item, Development Team, Ready - is there a reason why these are title cased inside the sentence?

If you are looking for emphasis - use **bold** or `tics` 

This is throughout the document

File: agile-development/backlog-management/refinement/readme.md [hunk 6]
```diff
@@ -16,22 +16,40 @@
...
+
 ## Facilitation Guidance
 
 How might one go about running the ceremony? Is there a general pattern that the ceremony should follow?
 
+- Product Owner: presents the ordered backlog and backlog list they'd like in the next iteration.
+- Process Lead: ensure the timebox.
+- Team members: asks questions and requests additional details on the backlog if needed.
```
Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:45]: nit: asks => ask and requests => request (as team members are plural)

**Review by TessFerrandez — APPROVED** [review]
Thank you for the contribution - just had a few language nits

File: agile-development/backlog-management/refinement/readme.md (hunk 3 above)
Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:19]:
```suggestion
Backlog refinement or formerly known as [backlog grooming](https://www.agilealliance.org/glossary/backlog-grooming/) is when the [product owner](https://www.agilealliance.org/glossary/product-owner/) and some, or all, of the rest of the team review items on the backlog to ensure the backlog contains the appropriate items, that they are prioritized, and that the items at the top of the backlog are ready for development. This activity occurs on a regular basis and may be an officially scheduled meeting or an ongoing activity.
```

File: agile-development/backlog-management/refinement/readme.md (hunk 4 above)
Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:31]:
```suggestion
The intent of backlog refinement is to ensure that the backlog keeps updating with items that are appropriate to their priority, i.e. higher priority items are with more details than lower priority ones. Also it is important that at any time there are enough user stories ready for planning in the next or even, where reasonable, the next two or three iterations but not too much further.
```

File: agile-development/backlog-management/refinement/readme.md (hunk 6 above)
Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:45]:
```suggestion
- Team members: ask questions and request additional details on the backlog if needed.
```

File: agile-development/backlog-management/refinement/readme.md (hunk 5 above)
Comment (TessFerrandez) [inline agile-development/backlog-management/refinement/readme.md:37]:
```suggestion
The expected outcome from the backlog refinement is all the team members are on the same page of what they are going to implement in the next or next few iterations with priority. A positive outcome of refinement is one where each backlog item can reasonably be “done” within the sprint time. Product backlog items that can be “done” by the development team within one sprint are deemed “ready” for selection in a sprint planning.
```

**Review by TessFerrandez — APPROVED** [review]
Changed some of the nits to suggestions

**Pull request merged**